modules/core/test/TCString.test.ts

```diff
@@ -6,8 +6,101 @@ import {
   GVL,
 } from '../src';
 
+type MinimalModelOptions = {
+  versions?: Number
+}
+
+const doEncode = (options = {}) => {
+
+  const tcString = new TCString();
+  const model = getMinimalModel(options);
+  return tcString.encode(model);
+
+}
+
+const truncDate = (date: Date) => Math.round(date.getTime() / 100) * 100;
+
+const getMinimalModel = (customOptions: MinimalModelOptions = {}) => {
+  const options = {
+    version: 2,
+    ...customOptions
+  };
+
+  const vendorlistJson = require('../dev/vendorlist.json');
+  const gvl: GVL = new GVL(vendorlistJson);
+  
+  const model = new TCModel(gvl);
+  model.cmpId = 123;
+  model.cmpVersion = 1;
+  model.consentLanguage = 'en';
+  model.version = options.version;
+  return model;
+
+};
+
+const compareModels = (actual: TCModel, expected: TCModel) => {
+  
+  const compareProp = (propName: string, transform = _ => _) => {
+    
+    const actualValue = transform(actual[propName]);
+    const expectedValue = transform(expected[propName]);
+    expect(actualValue, `Error while comparing '${propName}'`).to.eq(expectedValue);
+
+  }
+
+  compareProp('cmpId');
+  compareProp('cmpVersion');
+  compareProp('consentLanguage');
+  compareProp('consentScreen');
+  compareProp('created', truncDate);
+  compareProp('lastUpdated', truncDate);
+  compareProp('policyVersion');
+  compareProp('useNonStandardStacks');
+  compareProp('version');
+
+}
+
 describe('TCString', (): void => {
 
-  // soon...
+  describe('.encode(tcModel)', (): void => {
+    
+    it('does not throw if a valid TCModel is passed', (): void => {
+      
+      expect(doEncode).not.to.throw();
+
+    });
+
+    it('returns a string with 4 segments when TCModel version is 2', () => {
+
+      const encodedString = doEncode();
+
+      expect(encodedString.split('.')).to.be.lengthOf(4);
+
+    });
+
+    xit('WIP: currently fails. returns a string with only 1 segment when tCModel version is 1', () => {
```

Comment:
I'd like to remove this, I don't think it adds much to have a WIP test.  I'm unsure if encoding v1 strings is going to be in scope or not.  Perhaps a discussion for the transition subgroup.

Reply:
I agree